app/src/main/java/com/github/se/polyfit/ui/flow/AddMealFlow.kt

```diff
@@ -0,0 +1,36 @@
+package com.github.se.polyfit.ui.flow
+
+import androidx.compose.runtime.Composable
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
+import com.github.se.polyfit.ui.navigation.Navigation
+import com.github.se.polyfit.ui.navigation.Route
+import com.github.se.polyfit.ui.screen.IngredientScreen
+import com.github.se.polyfit.ui.screen.NutritionScreen
+import com.github.se.polyfit.viewmodel.meal.MealViewModel
+
+@Composable
+fun AddMealFlow(
+    mainNavigation: Navigation,
+    userID: String,
+    mealViewModel: MealViewModel = MealViewModel(userID)
+) {
+  val navController = rememberNavController()
+  val navigation = Navigation(navController)
+
+  NavHost(navController = navController, startDestination = Route.Ingredients) {
+    composable(Route.Ingredients) {
+      IngredientScreen(
+          mealViewModel = mealViewModel,
+          navigateBack = { mainNavigation.goBack() },
+          navigateForward = { navigation.navigateToNutrition() })
+    }
+    composable(Route.Nutrition) {
+      NutritionScreen(
+          mealViewModel = mealViewModel,
+          navigateBack = { navigation.goBack() },
+          navigateForward = { mainNavigation.navigateToHome() })
```

Comment:
Same, but i wouldn't direct commit through github review (for all) 😄 

```suggestion
          navigateBack = { goBack },
          navigateForward = { navigateToHome })
```